Snowplow/Internal/Session/SPSession.m

```diff
@@ -203,10 +210,11 @@ - (BOOL)shouldUpdateSession {
     return now < lastAccess || now - lastAccess > timeout;
 }
 
-- (void)updateSessionWithEventId:(NSString *)eventId {
+- (void)updateSessionWithEventId:(NSString *)eventId eventTimestamp: (long long)eventTimestamp {
     _isNewSession = NO;
     NSInteger sessionIndex = (_state.sessionIndex ?: 0) + 1;
-    _state = [[SPSessionState alloc] initWithFirstEventId:eventId currentSessionId:[SPUtilities getUUIDString] previousSessionId:_state.sessionId sessionIndex:sessionIndex userId:_userId storage:@"LOCAL_STORAGE"];
+    NSString *eventISOTimestamp = [SPUtilities timestampToISOString:eventTimestamp];
```

Comment:
Wondering if we need to add a new method on SPUtilities.
I wanted to reduce/remove the need of SPUtilities as it was almost a container of everything that didn't have a home. I don't know if there is any valuable alternative, otherwise just leave it there.

Reply:
I agree that it would be nice to refactor SPUtilities. Maybe separate the utility functions by type – e.g., date utilities, app info utilities. But I like having the helper functions separate from the main logic. For me, it makes it easier to read the code if less important stuff is separated out. I'm not sure if putting the function in `SPSession` would be a better choice here, it seems to me that it doesn't belong with Session logic..